### PR TITLE
feat : Real time calculation base on service in BOM

### PR DESCRIPTION
**Pull request by FHenry**
#25242 25242

### CODE REVIEW

File: htdocs/bom/tpl/objectline_view.tpl.php [hunk 1]
```diff
@@ -149,19 +149,28 @@
 	echo $line->efficiency;
 	print '</td>';
 } else {
-	// Unit
-	print '<td class="linecolunit nowrap right">';
+	// Duration
+	print '<td class="linecolqty nowrap right">';
```
Comment (eldy) [inline htdocs/bom/tpl/objectline_view.tpl.php:153]: linecolduration instead of linecoqty ?

File: htdocs/bom/tpl/objectline_view.tpl.php [hunk 2]
```diff
@@ -149,19 +149,28 @@
...
+			setEventMessages(null, $tmpproduct->errors, 'errors');
+		}
+		$prodDurationSecond = $prodDurationHours*3600;
+		print convertSecondToTime($line->qty*$prodDurationSecond, 'allhourmin');
 	}
 
 	print '</td>';
 
+	// Qty
+	print '<td class="linecolduration nowrap right">';
```
Comment (eldy) [inline htdocs/bom/tpl/objectline_view.tpl.php:169]: linecolqty instead of linecolduration ?

**Comment by eldy** [comment]
Something is hurting me here.
When we define a BOM, for example, duration of a service for the BOM is 2 days., it must remains 2 days as long as we do not change this. Even if the unit of service is changed from 1 min into 1 hour, the BOM must keep its reference of 2 days. Here you remove the unit of BOM line, this means that if we change a product unit reference, it have effect on all BOM that use this product and qty means nothing anymore.
The unit should remain on BOM line (because it is a duration entered by the end user, so a use rinput), and it is when we convert the BOM into the MO that we can convert it into duration in seconds (because it is a calculated duration). We can use for this the unit of BOM + Unit of the service).

**Comment by eldy** [comment]
So can you paste the screenshot of a BOM card after your change ?

**Comment by FHenry** [comment]
@eldy 
BOM
image

OF
image

**Pull request closed without merging**